internal/version/version.go

```diff
@@ -129,6 +130,44 @@ func VersionString() string {
 	return version
 }
 
+// BuildInfo contains version information
+type BuildInfo struct {
+	Major         string `json:"major,omitempty"`
+	Minor         string `json:"minor,omitempty"`
+	Version       string `json:"version"`
+	GitCommit     string `json:"gitCommit"`
+	GitTreeState  string `json:"gitTreeState"`
+	BuildDate     string `json:"buildDate"`
+	GoVersion     string `json:"goVersion"`
+	Compiler      string `json:"compiler"`
+	Platform      string `json:"platform"`
+}
+
+// GetBuildInfo returns detailed build information
+func GetBuildInfo() *BuildInfo {
+	info := &BuildInfo{
+		Version:       version,
+		GoVersion:     runtime.Version(),
+		Compiler:      runtime.Compiler,
+		Platform:      fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
+		GitTreeState:  "clean",
+		BuildDate:     time.Now().UTC().Format(time.RFC3339),
+	}
+
+	// Parse version into major.minor
+	if v, err := semver.ParseTolerant(version); err == nil {
+		info.Major = fmt.Sprintf("%d", v.Major)
+		info.Minor = fmt.Sprintf("%d", v.Minor)
+
+		// If there's a pre-release or metadata, it might contain the git commit
+		if len(v.Pre) > 0 {
+			info.GitCommit = v.Pre[0].VersionStr
+		}
```

Comment:
It doesn't seem to work on my environment:

```bash

$ make kubectl-gadget

$ ./kubectl-gadget version
Error: getting running Inspektor Gadget instances: Get "https://192.168.49.2:8443/apis/apps/v1/daemonsets?fieldSelector=metadata.name%3Dgadget&labelSelector=k8s-app%3Dgadget": dial tcp 192.168.49.2:8443: connect: no route to host
Client Version: v0.40.0-130-gad3cb347e-dirty
  Major: 0
  Minor: 40
  Git Commit: 130-gad3cb347e-dirty
  Git Tree State: clean
  Build Date: 2025-05-26T16:04:54Z
  Go Version: go1.24.0
  Compiler: gc
  Platform: linux/amd64
```

`Git Commit: 130-gad3cb347e-dirty` is incorrect.